src/test/java/org/cactoos/io/TempFileTest.java

```diff
@@ -0,0 +1,110 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2018 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.io;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.cactoos.func.MatcherOf;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+
+/**
+ * Unit tests for {@link TempFile}.
+ *
+ * @author George Aristy (george.aristy@gmail.com)
+ * @version $Id$
+ * @since 1.0
+ * @checkstyle JavadocMethodCheck (500 lines)
+ */
+public final class TempFileTest {
+
+    @Test
+    public void tmpFileIsCreated() throws Exception {
+        try (final TempFile temp = new TempFile()) {
+            MatcherAssert.assertThat(
+                "Cannot create a temp file",
+                Files.exists(temp.value()),
+                Matchers.is(true)
+            );
+        }
+    }
+
+    @Test
+    public void createFileInCustomPath() throws Exception {
+        final Path custom = Paths.get(System.getProperty("user.home"));
+        try (final TempFile temp = new TempFile(() -> custom, "", "")) {
+            MatcherAssert.assertThat(
+                "Cannot create a temp file at a custom path",
+                temp,
+                Matchers.allOf(
+                    new MatcherOf<>(
+                        t -> { return Files.exists(t.value()); }
+                    ),
+                    new MatcherOf<>(
+                        t -> { return t.value().getParent().equals(custom); }
+                    )
+                )
+            );
+        }
+    }
+
+    @Test
+    public void closedFileCeasesToExist() throws Exception {
```

Comment:
@llorllale Why does a closed file cease to exist? Can't it be reopened? It doesn't seem logical to me.

Reply:
I view temporary files as *ephemeral*. They are for 'fire and forget' scenarios. Don't you agree?

Reply:
I see... valid point, but pls make it clear in the JavaDoc :)